src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -63,13 +63,20 @@ type ByocAws struct {
 
 var _ client.Provider = (*ByocAws)(nil)
 
-func NewByocProvider(ctx context.Context, grpcClient client.GrpcClient, tenantId types.TenantID) *ByocAws {
+var ErrMissingAwsCreds error = errors.New("AWS credentials must be set (https://docs.defang.io/docs/providers/aws/#getting-started)")
+
+func NewByocProvider(ctx context.Context, grpcClient client.GrpcClient, tenantId types.TenantID) (*ByocAws, error) {
 	b := &ByocAws{
 		cdTasks: make(map[string]ecs.TaskArn),
 		driver:  cfn.New(byoc.CdTaskPrefix, aws.Region("")), // default region
 	}
 	b.ByocBaseClient = byoc.NewByocBaseClient(ctx, grpcClient, tenantId, b)
-	return b
+
+	_, err := b.AccountInfo(ctx)
+	if err != nil {
```

Comment:
are all errors related to missing credentials?

Reply:
yes, but we can wrap the err to preserve it, which will probably be a better practice